### PR TITLE
Cldc 576 reproduce about this log

### PR DESCRIPTION
**Pull request by magicmilo**

### CODE REVIEW

File: app/helpers/check_answers_helper.rb [hunk 1]
```diff
@@ -28,6 +28,7 @@ def total_questions(subsection, case_log, form)
   def filter_conditional_questions(questions, case_log)
     applicable_questions = questions
 
+    # puts applicable_questions.count
```
Comment (baarkerlounger) [inline app/helpers/check_answers_helper.rb]: This looks like it was for debugging?

File: app/views/form/_select_question.html.erb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+
+<%= answers = question["answer_options"].map {|key, value| OpenStruct.new(id:key, name: value)}
```
Comment (baarkerlounger) [inline app/views/form/_select_question.html.erb:2]: Needs linting I think

**Comment by baarkerlounger** [comment]
Am I right in thinking this PR just bring in the about this log pages, it doesn't yet enable complex routing? (i.e. things like, if it's a relet 4-5 pages become unnecessary to show?)

File: app/helpers/check_answers_helper.rb [hunk 3]
```diff
@@ -56,11 +56,12 @@ def condition_not_met(case_log, question_key, question, condition)
...
       operand = condition[/\d+/].to_i
       case_log[question_key].blank? || !case_log[question_key].send(operator, operand)
+    when "text"
+      case_log[question_key].blank? || !condition.include?(case_log[question_key])
     when "radio"
       case_log[question_key].blank? || !condition.include?(case_log[question_key])
     else
-      raise "Not implemented yet"
-    end
+      raise "Not implemented yet" end
```
Comment (baarkerlounger) [inline app/helpers/check_answers_helper.rb]: Don't think you need the `end` here do you?

File: app/views/form/_select_question.html.erb (hunk 2 above)
Comment (baarkerlounger) [reply app/views/form/_select_question.html.erb:2]: (Spaces on the `}`)

File: app/helpers/check_answers_helper.rb (hunk 1 above)
Comment (magicmilo) [reply app/helpers/check_answers_helper.rb]: Yep sorry got it

> Am I right in thinking this PR just bring in the about this log pages, it doesn't yet enable complex routing? (i.e. things like, if it's a relet 4-5 pages become unnecessary to show?)

It's to prove that the routing we have already is capable of any level of complexity

File: app/views/form/_select_question.html.erb (hunk 2 above)
Comment (magicmilo) [reply app/views/form/_select_question.html.erb:2]: ran rubocop on it. Didn't show any errors.

File: app/helpers/check_answers_helper.rb (hunk 3 above)
Comment (magicmilo) [reply app/helpers/check_answers_helper.rb]: Do need it but moved it onto correct line

File: app/views/form/_select_question.html.erb (hunk 2 above)
Comment (baarkerlounger) [reply app/views/form/_select_question.html.erb:2]: Interesting, wonder if it doesn't lint erbs

**Comment by magicmilo** [comment]
> Am I right in thinking this PR just bring in the about this log pages, it doesn't yet enable complex routing? (i.e. things like, if it's a relet 4-5 pages become unnecessary to show?)

It is really just to prove that the logic we have works. We already have complex routing. Any scenario you can think of I'd be happy to prove that.

**Review by baarkerlounger — APPROVED** [review]

**Pull request merged**